core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/codeGen/MarkersExtractor.kt

```diff
@@ -17,14 +17,33 @@ import kotlin.reflect.full.withNullability
 import kotlin.reflect.jvm.jvmErasure
 import kotlin.reflect.typeOf
 
-internal fun KType.shouldBeConvertedToFrameColumn(): Boolean = when (jvmErasure) {
-    DataFrame::class -> true
-    List::class -> arguments[0].type?.jvmErasure?.hasAnnotation<DataSchema>() == true
-    else -> false
+internal fun KType.getFieldKind(): FieldKind = when {
+    jvmErasure == DataFrame::class -> Frame
+    jvmErasure == List::class && (arguments[0].type?.jvmErasure?.hasAnnotation<DataSchema>() == true) -> ListToFrame
```

Comment:
Could be here only List type, is it possible to meet a Map, Array?

Reply:
Only `List`, i.e. given
```
@DataSchema
data class Record(val a: String)
```

`data class MyClass(val l: LIst<Record>)` and `data class MyClass(val l: DataFrame<Record>)`
are equivalent declarations from `toDataFrame` and `MarkersExtractor.get` point of view.